kratos/spatial_containers/geometrical_objects_bins.cpp

```diff
@@ -158,6 +158,91 @@ GeometricalObjectsBins::ResultType GeometricalObjectsBins::SearchNearestInRadius
 /***********************************************************************************/
 /***********************************************************************************/
 
+void GeometricalObjectsBins::SearchInBoundingBox(
+    const PointType& rPoint,
+    const array_1d<double, 3>& rMinPoint,
+    const array_1d<double, 3>& rMaxPoint,
+    std::vector<ResultType>& rResults
+    )
+{
+    // Clear the results
+    rResults.clear();
+
+    // Define bounding box
+    BoundingBox<PointType> bounding_box;
+    noalias(bounding_box.GetMinPoint().Coordinates()) = rMinPoint;
+    noalias(bounding_box.GetMaxPoint().Coordinates()) = rMaxPoint;
+
+    // Initialize the current size
+    std::size_t current_size = 0;
+
+    // Initialize the candidates
+    std::unordered_set<GeometricalObject*> candidates;
+    std::unordered_set<GeometricalObject*> to_remove;
+
+    // Initialize the position bounds
+    array_1d<std::size_t, Dimension> min_position;
+    array_1d<std::size_t, Dimension> max_position;
+
+    // Calculate the position bounds
+    for(unsigned int i = 0; i < Dimension; i++ ) {
+        min_position[i] = CalculatePosition(rMinPoint[i], i);
+        max_position[i] = CalculatePosition(rMaxPoint[i], i) + 1;
+    }
+
+    // Loop over the cells and gather candidates
+    for(std::size_t k = min_position[2]; k < max_position[2]; k++) {
+        for(std::size_t j = min_position[1]; j < max_position[1]; j++) {
+            for(std::size_t i = min_position[0]; i < max_position[0]; i++) {
+                auto& r_cell = GetCell(i, j, k);
+                if (IsCellBoundingBoxInsideBoundingBox(i, j, k, rPoint, bounding_box)) {
+                    current_size = rResults.size();
+                    rResults.reserve(current_size + r_cell.size());
+                    for(auto p_geometrical_object : r_cell) {
+                        auto insert_result = to_remove.insert(p_geometrical_object);
+                        if (insert_result.second) {
+                            auto& r_geometry = p_geometrical_object->GetGeometry();
+                            const double distance = r_geometry.CalculateDistance(rPoint, mTolerance);
```

Comment:
I would not calculate the distance in this method. Here the distance calculation cost is much more than the entire search

Reply:
Okay